src/main/java/com/infusionsoft/dataflow/templates/DeleteExpiredHostedEmails.java

```diff
@@ -0,0 +1,81 @@
+package com.infusionsoft.dataflow.templates;
+
+import static com.infusionsoft.dataflow.utils.ZonedDateTimeUtils.UTC;
+
+import com.google.cloud.teleport.templates.common.PubsubConverters.PubsubReadOptions;
+import com.google.datastore.v1.Entity;
+import com.google.datastore.v1.Key;
+import com.infusionsoft.dataflow.utils.ZonedDateTimeUtils;
+import java.time.ZonedDateTime;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.io.gcp.datastore.DatastoreIO;
+import org.apache.beam.sdk.options.Description;
+import org.apache.beam.sdk.options.PipelineOptions;
+import org.apache.beam.sdk.options.PipelineOptionsFactory;
+import org.apache.beam.sdk.options.StreamingOptions;
+import org.apache.beam.sdk.options.ValueProvider;
+import org.apache.beam.sdk.options.ValueProvider.NestedValueProvider;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.transforms.Reshuffle;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+
+/**
+ * A template that deletes expired hosted emails.
+ *
+ * Used by hosted-email-api
+ *
+ * Deploy to sand:
+ * mvn compile exec:java -Dexec.mainClass=com.infusionsoft.dataflow.templates.DeleteExpiredHostedEmails -Dexec.args="--project=is-hosted-email-api-sand --stagingLocation=gs://dataflow-is-hosted-email-api-sand/staging --templateLocation=gs://dataflow-is-hosted-email-api-sand/templates/delete_expired_emails --runner=DataflowRunner --serviceAccount=is-hosted-email-api-sand@appspot.gserviceaccount.com --datastoreProjectId=is-hosted-email-api-sand"
```

Comment:
I sense some degree of pain was experienced to reach this happy state.  Glad to see some javadocs to mark the way:)

Reply:
yes. yes, indeed. :)